pkg/odo/cli/component/watch.go

```diff
@@ -86,18 +86,18 @@ func (wo *WatchOptions) Complete(name string, cmd *cobra.Command, args []string)
 // Validate validates the watch parameters
 func (wo *WatchOptions) Validate() (err error) {
 
-	// Validate component path existence and accessibility permissions for odo
-	if _, err := os.Stat(wo.sourcePath); err != nil {
-		return errors.Wrapf(err, "Cannot watch %s", wo.sourcePath)
-	}
-
 	// Validate source of component is either local source or binary path until git watch is supported
 	if wo.sourceType != "binary" && wo.sourceType != "local" {
 		return fmt.Errorf("Watch is supported by binary and local components only and source type of component %s is %s",
 			wo.localConfig.GetName(),
 			wo.sourceType)
 	}
 
+	// Validate component path existence and accessibility permissions for odo
+	if _, err := os.Stat(wo.sourcePath); err != nil {
```

Comment:
Just thinking out loud here. Can we not have a simpler condition that throws this error if the `sourceType` is `git`? If not, then I would recommend that we add more comments here about when the said error could hit us and why we're using this way to handle it.

Reply:
@dharmit the line
https://github.com/openshift/odo/pull/2280/files/0a1625b32e96ffa0ba608170b113ec37bbae6baf#diff-f36b01078d99cd02f0a8239f8d2553ebR90 is doing the check for `sourceType != git` validation. This check is if the user tries to watch a path that doesn't exist or/ we dont have permissions to watch it.